lib/entities/resource-provider.ts

```diff
@@ -92,6 +105,44 @@ function createResourceProviderApi(makeRequest: MakeRequest) {
         params: getParams(data),
       })
     },
+
+    getResourceType: function getResourceType(id: string) {
```

Comment:
Is it a common pattern in this SDK to expose sub resources both in parent resource methods and standalone? Also, do we need docs for these methods?

Reply:
there is a separate ticket for updating the docs after we finish this one

Reply:
for the other question about the standalone vs parent resource in the legacy client, usually in normal cases the parent resource has create and get methods and the standalone resource (child resource) has the update and delete. Let me give you an example to make it clearer: if you want to create a new `appDefinition`, you do it through the `organization` resource, so you init the client, `getOrganisation`, and then through the organisation entity you do the `createAppDefition` so you end up with the AppDefintion entity that you can update or delete. You cannot add a get on that standalone entity because you can't call that method on the entity itself. however, in the plain client you can do it because basically you call everything related to the standalone entity on the standalone entity.